webapp/channels/src/components/more_direct_channels/list/list.tsx

```diff
@@ -36,108 +41,160 @@ type Props = {
+            },
+            [props.currentUserId],
+        );
 
-    let note;
-    if (props.isExistingChannel) {
-        if (props.values.length >= MAX_SELECTABLE_VALUES) {
-            note = (
-                <FormattedMessage
-                    id='more_direct_channels.new_convo_note.full'
-                    defaultMessage={'You\'ve reached the maximum number of people for this conversation. Consider creating a private channel instead.'}
-                />
-            );
-        } else {
-            note = (
-                <FormattedMessage
-                    id='more_direct_channels.new_convo_note'
-                    defaultMessage={'This will start a new conversation. If you\'re adding a lot of people, consider creating a private channel instead.'}
-                />
+        const handleCreateChannel = () => {
+            props.handleHide();
+            dispatch(
+                openModal({
+                    modalId: ModalIdentifiers.NEW_CHANNEL_MODAL,
+                    dialogType: NewChannelModal,
+                }),
             );
+        };
+
+        const intl = useIntl();
+
+        let note;
+        if (props.isExistingChannel) {
+            if (props.values.length >= MAX_SELECTABLE_VALUES) {
+                note = (
+                    <FormattedMessage
+                        id='more_direct_channels.new_convo_note.full'
+                        defaultMessage={
+                            "You've reached the maximum number of people for this conversation. Consider creating a private channel instead."
+                        }
+                    />
+                );
+            } else {
+                note = (
+                    <FormattedMessage
+                        id='more_direct_channels.new_convo_note'
+                        defaultMessage={
+                            "This will start a new conversation. If you're adding a lot of people, consider creating a private channel instead."
+                        }
+                    />
+                );
+            }
         }
-    }
-
-    const options = useMemo(() => {
-        return props.options.map(optionValue);
-    }, [props.options]);
-
-    return (
-        <MultiSelect<OptionValue>
-            ref={ref}
-            options={options}
-            optionRenderer={renderOptionValue}
-            intl={intl}
-            selectedItemRef={props.selectedItemRef}
-            values={props.values}
-            valueRenderer={renderValue}
-            ariaLabelRenderer={renderAriaLabel}
-            perPage={USERS_PER_PAGE}
-            handlePageChange={props.handlePageChange}
-            handleInput={props.search}
-            handleDelete={props.handleDelete}
-            handleAdd={props.addValue}
-            handleSubmit={props.handleSubmit}
-            noteText={note}
-            maxValues={MAX_SELECTABLE_VALUES}
-            numRemainingText={
+
+        const RemainingText = () => {
```

Comment:
Defining this here sort of as a component within a component looks a bit odd to me because it's accessing the props and context of the outer component. It feels like something that may break at some point.

Would you mind changing this so that it's not a functional component? Something like
```js
let remainingText;
if (props.values.length >= MAX_SELECTABLE_VALUES) {
   remainingtext = ...;
} else {
   remainingtext = ...;
}
```
should work the same, but it wouldn't share props between the components

Reply:
@hmhealey  Thank you for your review. I fixed it. Could you review it?